app/src/main/java/com/orange/ods/app/ui/search/SearchScreen.kt

```diff
@@ -16,41 +16,157 @@ import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.ui.text.input.TextFieldValue
+import com.orange.ods.app.R
+import com.orange.ods.app.ui.LocalOdsGuideline
+import com.orange.ods.app.ui.components.Component
+import com.orange.ods.app.ui.components.Variant
 import com.orange.ods.app.ui.components.components
+import com.orange.ods.app.ui.guidelines.color.DialogColor
+import com.orange.ods.app.ui.guidelines.color.getValue
+import com.orange.ods.app.ui.guidelines.spacing.Spacing
+import com.orange.ods.app.ui.utilities.DrawableManager
 import com.orange.ods.compose.component.list.OdsListItem
 import com.orange.ods.compose.component.list.OdsListItemIcon
 import com.orange.ods.compose.component.list.OdsListItemIconType
 import com.orange.ods.compose.component.list.iconType
+import com.orange.ods.theme.guideline.GuidelineColor
+import com.orange.ods.theme.guideline.toHexString
 import com.orange.ods.utilities.extension.orElse
 
 @Composable
-fun SearchScreen(searchedText: MutableState<TextFieldValue>, onComponentClick: (Long) -> Unit) {
-    val filterComponents = components.filter { component ->
-        searchedText.value.text.isEmpty() || stringResource(id = component.titleRes).lowercase().contains(searchedText.value.text.lowercase())
+fun SearchScreen(
+    searchedText: MutableState<TextFieldValue>,
+    onComponentClick: (Long) -> Unit,
+    onVariantClick: (Long) -> Unit,
+    onGuidelineSpacingClick: (Long) -> Unit
+) {
+
+    val context = LocalContext.current
+
+    val filteredComponents = components.filter { component ->
+        searchedText.value.text.isEmpty() || stringResource(id = component.titleRes).lowercase()
+            .contains(searchedText.value.text.lowercase())
+    }.asSequence()
+
+    val filteredSpacings = Spacing.values().filter { spacing ->
+        searchedText.value.text.isEmpty() || spacing.tokenName.lowercase()
+            .contains(searchedText.value.text.lowercase())
+    }
+
+    val filteredGuidelineColors = LocalOdsGuideline.current.guidelineColors.filter { guidelineColor ->
+        searchedText.value.text.isEmpty() || guidelineColor.getName().lowercase().contains(searchedText.value.text.lowercase()) ||
+                guidelineColor.lightThemeName.lowercase().contains(searchedText.value.text.lowercase()) ||
+                guidelineColor.darkThemeName.lowercase().contains(searchedText.value.text.lowercase())
     }
+
+    val filteredVariants = components.filter { it.variants.isNotEmpty() }
+        .flatMap { component ->
+            val componentImageRes = component.smallImageRes.orElse { component.imageRes }
+            component.variants.filter { variant ->
+                searchedText.value.text.isEmpty() || context.getString(variant.titleRes).lowercase()
+                    .contains(searchedText.value.text.lowercase())
+            }.map { variant ->
+                componentImageRes to variant
+            }
+        }
+
+    data class SearchResult(
+        val title: String,
+        val route: Long,
```

Comment:
This is not really the route but the id of the element. Could you rename it in `id`?